src/main/java/org/jenkinsci/backend/confluence/pageremover/webapp/PageRemoverResource.java

```diff
@@ -0,0 +1,180 @@
+import static org.jenkinsci.backend.confluence.pageremover.Spambot.BLACKLIST;
+
+@Path("/page-remover")
+@Produces(MediaType.TEXT_PLAIN)
+public class PageRemoverResource {
+    private final String space;
+    private final String mailRecipient;
+    private final String smtpServer;
+
+    public PageRemoverResource(String space, String mailRecipient, String smtpServer) {
+        this.space = space;
+        this.mailRecipient = mailRecipient;
+        this.smtpServer = smtpServer;
+    }
+
+    @POST
+    public Response submit(@QueryParam("subject") Optional<String> subject,
+                           @QueryParam("body-plain") Optional<String> body) throws Exception {
+
+        PageNotification pageNotification = PageNotification.parse(subject.or(""), body.or(""));
+
+        if (subject.or("").startsWith("Re: ") && body.isPresent()) {
+            removePage(subject.get(), body.get());
+            return Response.ok().build();
+        }
+
+        if (pageNotification != null) {
+            processNotification(pageNotification);
+            return Response.ok().build();
+        }
+
+        System.err.println("Not a confluence notification email: " + subject.or("[empty subject]"));
+        return Response.ok().build();
+
+    }
+
+    /**
+     * Processes notification email from Confluence, and take appropriate actions.
+     */
+    private void processNotification(PageNotification n) throws Exception {
+        System.err.println("Parsed "+n);
+
+        if (n.action.equals("added")) {
+            Connection con = new Connection();
+            RemotePage p = con.getPage(n.spaceID, n.pageTitle);
+            Language lang = null;
+            try {
+                lang = new LanguageDetection().detect(p.getContent());
+                if (lang.lang.equals("id") && lang.prob>0.99) {
+                    // highly confident that this is a spam. go ahead and remove it
+                    removePage(con, p);
+                    banAccount(n.authorID);
+                    return;
+                }
+            } catch (LangDetectException e) {
+                System.err.println("Failed to detect language");
+                e.printStackTrace();
+            }
+
+            if (BLACKLIST.matches(p.getContent()) || BLACKLIST.matches(p.getTitle())) {
+                removePage(con, p);
+                banAccount(n.authorID);
+                return;
+            }
+
+            String body = String.format("Language detection: %s\nWiki: %s\n\n\nSee https://github.com/jenkinsci/backend-confluence-spam-remover about this bot", lang, n);
+            System.err.println(body);
+
+            sendResponse(String.format("Keeping page %s", p.getTitle()), body);
+        }
+    }
+
+    /**
+     * Ban the user.
+     */
+    private void banAccount(String id) {
+        try {
+            AccountServer app = new AccountServer(ConfigurationLoader.from(new File("./config.properties")).as(Config.class));
```

Comment:
Won't this result in `config.properties` being read over and over and over again?